### ISSUE TITLE
WordPress REST API not purged on new post or updates

### ISSUE THREAD

**Issue by rmens** [cloudflare/Cloudflare-WordPress#248]
Hi,

I'm using the Cloudflare WordPress plug-in to automatically cache everything on a very high traffic WordPress site and app. The plug-in does a great job purging the cache on creation of a new post and edits, but the REST API doesn't seem to be purged.

I hard-coded a small plug-in to purge the following routes in the Cloudflare API on post update or publish, but this seems to be something that the Cloudflare plug-in should do.

```https://domain/wp-json/wp/v2/posts/_the_post_id```
```https://domain/wp-json/wp/v2/posts```

**Comment by AJB99** [comment]
I've run into this same issue using WP as a headless CMS with a Vue.js frontend that makes XHR requests to `https://wp.mysite.com/wp-json/*` to fetch post, page, media, etc data.

I'm attempting to use a Cloudflare Page Rule to cache all responses from requests made to `wp.mysite.com/wp-json/*`, but the Cloudflare Wordpress plugin is failing to purge the cache for that Page Rule when any content (post, page, media, etc.) is updated via my WP Admin Console.

These are my Page Rule settings:
<img width="781" alt="image" src="https://user-images.githubusercontent.com/6812084/80859516-a3579900-8c2f-11ea-9bb9-269f41df08e5.png">

If I remove the "Edge Cache TTL" setting then I'm seeing that Cloudflare always sends the request to my source server. But if I enable it (and it *does* cache responses from the `/wp-json/*` URL) then the cache is not purged via the Cloudflare WP Plugin when I make updates to my content.

A response from @Cloudflare-Wordpress would be appreciated. This is a significant limitation that Cloudflare needs to address for customers looking to implement a headless WP CMS and use Cloudflare to make it reasonably fast.  Headless WP essentially *requires* Cloudflare (Read: $$$$).

**Comment by garrettgalow** [comment]
Unfortunately, this functionality isn't supported by our plugin. The plugin purges cache on updates to the style of the site like theme changes. The plugin does not purge .html content, given that our default behavior is not to cache such content.

**Comment by jabranr** [comment]
@garrettgalow is this something to be considered as potential new feature? I might be able to look into this and send a PR.

**Comment by rmens** [comment]
@jabranr A PR would be nice anyway. This plug-in is not updated that often so running a fork that integrates this is doable. If it's merged switching back to the official plug-in is always an option.

**Comment by garrettgalow** [comment]
I believe we have something in the works that is going to address this. However, the plugin by default will not support this, as its not built to handle all content being cached.

**Comment by markjaquith** [comment]
This issue could use a second look now that APO is in the picture.

**Comment by rmens** [comment]
Yes. I wonder if @garrettgalow was referring to APO as what was in the works.

**Comment by rmens** [comment]
Still curious if this is fixed by enabling APO.

**Comment by davidpelayo** [comment]
Is there any PR or activity that aims to fix this issue anytime soon?

**Comment by alyxb** [comment]
Has this been resolved?

**Comment by rmens** [comment]
Not that I'm aware of. This might be fixed with APO, but I don't use that so if someone who uses APO could test it that would be great.

**Comment by physixfan** [comment]
I can confirm we have the same issue: even with the new APO feature, RESI API cache are not purged properly after post updating. We hope some can make a PR or something to resolve this issue.

**Comment by physixfan** [comment]
> @garrettgalow is this something to be considered as potential new feature? I might be able to look into this and send a PR.

We would really appreciate your effort if you can add this as a new feature!

**Comment by physixfan** [comment]
The plugin used to have a feature called "Auto Purge Content On Update" and it worked properly. I am wondering why it is disabled after APO feature is introduced. Can someone in CloudFlare fix this problem?

**Comment by Jineshpanchal** [comment]
We have ben facing this issue since more than few month but not yet address.

https://community.cloudflare.com/t/since-past-2-days-pages-are-not-updated-and-served-from-cached-even-after-clearing-cache/411786

https://support.cloudflare.com/hc/en-us/requests/2530764?page=1

**Comment by johnaps94** [comment]
I am having the same problem, no page purge when updating content on site...
Tried many things, nothing triggers it even after an hour is passed...

Do you have (for example) a specific product page cache purged when you updated that product's price?

**Comment by bsahins** [comment]
same here

**Comment by rmens** [comment]
Since this issue has many thumbs up and eyes I'll try to ask again about if APO fixes this. I only have one Wordpress site on a professional plan and am not willing to take the risk of testing this in production so an answer would be appreciated. 

In more general terms I'm curious if there is a product manager for APO or WordPress products at Cloudflare. This plug-in seems to get drive-by updates but I haven't seem it improve in meaningful ways the past years. It's odd that the biggest CMS in the world isn't begin serviced well by Cloudflare.

**Comment by bsahins** [comment]
I don't think anyone from Cloudflare read our comments in here.

**Comment by rmens** [comment]
@bsahins Sorry i’m in publishing. It only works for posts not for e-commerce.

**Comment by bsahins** [comment]
@rmens thanks for quick feedback. Actually products are also posts in woocommerce. If your code is working for posts, it will work for products too.
It would be great if you can share, I can test and get back to you.

**Comment by rmens** [comment]
I uploaded the plug-in I use as code snippet on https://gist.github.com/rmens/3fde8d6694418e8038ca4f623a187d3c. Let's not pollute the discussion further here.

**Comment by rmens** [comment]
Since this issue has garnered much eyes and thumbs, I'll ask again: Does APO fix this? I have one WordPress site on a professional plan and can't risk testing it in production, so I'd appreciate an answer.

More broadly, is there a product manager for APO or WordPress products at Cloudflare? The plugin gets occasional updates, but I haven't seen significant improvements over the years. It's strange that Cloudflare doesn't better support the world's largest CMS.

**Comment by ned-bs** [comment]
same for product price and stock updates. They just don't care.

**Comment by ned-bs** [comment]
5 years and no reply.

**Comment by ned-bs** [comment]
6 years and no reply

**Comment by rmens** [comment]
Yes it’s a shame that the worlds most popular CMS isn’t treated as a first class citizen on Cloudflare.
